gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetup.java

```diff
@@ -0,0 +1,109 @@
+ * certificates specified by their serialNumbers & alias name from {@code certsDir} into the JDK's truststore.
+ * A certificate will be imported iff the serial number already exists in the truststore.
+ * The class will be called by the Gradle setup script in
+ * <a href="file:../resources/gradle-jdks-setup.sh">resources/gradle-jdks-setup.sh</a>.
+ */
+public final class GradleJdkInstallationSetup {
+
+    public static void main(String[] args) throws IOException {
+        StdLogger logger = new StdLogger();
+        if (args.length != 2) {
+            throw new IllegalArgumentException("Expected two arguments: destinationJdkInstallationDir and certsDir");
+        }
+        Path destinationJdkInstallationDir = Path.of(args[0]);
+        Path certsDir = Path.of(args[1]);
+        atomicCopyJdkInstallationDirectory(logger, destinationJdkInstallationDir);
+        Map<String, String> certNamesToSerialNumbers = extractCertsSerialNumbers(logger, certsDir);
+        CaResources.maybeImportCertsInJdk(logger, destinationJdkInstallationDir, certNamesToSerialNumbers);
+    }
+
+    private static void atomicCopyJdkInstallationDirectory(ILogger logger, Path destinationJdkInstallationDirectory)
+            throws IOException {
+        Path currentJavaHome = Path.of(System.getProperty("java.home"));
+        Path jdksInstallationDirectory = destinationJdkInstallationDirectory.getParent();
+        if (!jdksInstallationDirectory.toFile().exists()) {
+            Files.createDirectories(jdksInstallationDirectory);
+        }
+        Path lockFile = jdksInstallationDirectory.resolve(destinationJdkInstallationDirectory.getFileName() + ".lock");
+        try (FileChannel channel = FileChannel.open(lockFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
+            channel.lock();
+            // double-check, now that we hold the lock
+            if (Files.exists(destinationJdkInstallationDirectory)) {
+                logger.log(String.format("Distribution URL %s already exists", destinationJdkInstallationDirectory));
+                return;
+            }
+            logger.log(
+                    String.format("Copying JDK from %s into %s", currentJavaHome, destinationJdkInstallationDirectory));
+            FileUtils.copyDirectory(currentJavaHome, destinationJdkInstallationDirectory);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to acquire locks, won't move the JDK installation directory", e);
+        }
+    }
+
+    private static Map<String, String> extractCertsSerialNumbers(ILogger logger, Path certsDirectory)
+            throws IOException {
+        if (!Files.exists(certsDirectory)) {
+            logger.log("No certs directory found, skipping import of certificates");
+            return Map.of();
+        }
+        Map<String, String> certSerialNumbersToAliases = new HashMap<>();
+        Files.walkFileTree(certsDirectory, new FileVisitor<Path>() {
```

Comment:
I think we can just use `Files.list` here rather than walkFileTree - I think they're all going to be in the same dir?